# mlsttype

`MLSType` from [Enterobase Tool Kit (EToKi)](https://github.com/zheminzhou/EToKi). 

EToKi `MLSType` identities allelic sequences in a queried genome, by comparing it with the exemplar alleles generated by `MLSTdb`. It requires USEARCH (GPLv3) and BLAST (Public domain). Like the original package, this one is also released under GPLv3.


## Calculate cgMLST genotype for a queried genome and print to standard output

```bash
python mlstype/MLSType.py -i examples/SAL_QD2830AA_AS.result.fasta.gz -r examples/cgMLST_v2_ref.fasta.gz -k cgmlst_SAL -o STDOUT
```

### Citation

Zhou Z, Alikhan NF, Mohamed K, Fan Y; Agama Study Group; Achtman M. The EnteroBase user's guide, with case studies on Salmonella transmissions, Yersinia pestis phylogeny, and Escherichia core genomic diversity. Genome Res. 2020 Jan;30(1):138-152. doi: 10.1101/gr.251678.119. Epub 2019 Dec 6. PMID: 31809257; PMCID: PMC6961584.
